draft-vvv-webtransport-quic.md

```diff
@@ -237,21 +241,20 @@ The user agent MUST send the "Origin" field.  The "Origin" field MUST be set to
 the origin of the client initiating the connection, serialized as described in
 the "serializing a request origin" section of [FETCH].
 
+The server indicates that it accepts the origin by sending its header block and
+a FIN on the stream 0.  If the server does not accept the origin, it MUST close
+the connection.
```

Comment:
How does the client know that a connection close was due to an unaccepted origin? If it receives a CONNECTION_CLOSE prior to receiving a FIN on stream 0?

The problem is that there are many sources of CONNECTION_CLOSE errors. The server could close the connection due to an invalid path or it just decided to immediately terminate the connection. In either case, the client will receive a QUIC application close and will incorrectly assume that it was due to an invalid origin.

To correctly identify invalid origins, there should be an explicit and retransmitted signal. Either an "invalid origin" response message (on stream 0)  or CONNECTION_CLOSE error code.

Reply:
You could also abort the stream with a RESET_STREAM in this case. That way you wouldn't even need an error code (although it would help) in order to detect the case where the origin exchange is pass/fail.

Reply:
Yeah, `RESET_STREAM` would be nice with or without error codes. It's better than CONNECTION_CLOSE because it can be reliable and does not infringe upon the application error code space.

Reply:
Those are good points.  I wrote my thoughts on them in #15